adagrams/game.py

```diff
@@ -1,11 +1,103 @@
-def draw_letters():
-    pass
+import random
+LETTERS = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+letter_value =[
+{"A,E, I, O, U, L, N, R, S, T":1},
+{"D, G": 2},
+{"B, C, M, P": 3},
+{"F, H, V, W, Y": 4},
+{"K": 5},
+{"J, X": 8},
+{"Q, Z": 10}
+]
 
+def draw_letters():
+    letter_list = []
+    freq_dict = {}
+    count = 0
+    while count < 10:
+        letter = chr(random.randint(ord('A'), ord('Z')))
+        freq_dict[letter] = freq_dict.get(letter,0)+1
+    
+        if freq_dict[letter] > LETTERS[letter]:
+            continue
+        else:
+            letter_list.append(letter)
+        count += 1
+    return letter_list    
```

Comment:
Nice job using a dictionary to store the available letters, instead you could create a copy of the letter bank, iterate over it and check if the random letter's value is greater than 0 and if it is, append to the letter_list and use a helper dictionary to decrement the value in the letter pool to keep track of how many letters have been played! Something like this:
```
def draw_letters():
    letter_bank_copy = letter_bank[:]
    letters = []

    while len(letters) < 10:
        letter = random.choice(list(letter_pool_copy.keys()))
        if letter_pool_copy[letter] > 0:
            letters.append(letter)
            letter_pool_copy[letter] -= 1

    return letters
```

Reply:
I noticed you used a counter variable but rather than using a list of the letters in the hand, could we build a helper data structure (like a dictionary) that could let us look up if a letter was part of a user's hand?

If the helper dictionary had keys that were the letters and the values were the number of letters in the hand, then you could leverage the fact that key look up in dictionaries is a constant O(1) operation.

You can check if key (the letter in the hand) in helper_dictionary, which has constant look up time